bin/src/hickory-dns.rs

```diff
@@ -493,6 +524,17 @@ fn config_https(
     Ok(())
 }
 
+#[cfg(feature = "prometheus-metrics")]
+fn config_prometheus(socket: impl Into<SocketAddr>) -> Result<(), String> {
```

Comment:
Given that this is not actually async, I think we can remove the `runtime.block_on()` wrapper here, in which case I think we can just inline this into the single caller?

Reply:
The reason for the `runtime.block_on()` is to avoid that the prometheus endpoint will be spawning an additional tokio runtime when it cannot detect an existing one.

https://docs.rs/metrics-exporter-prometheus/latest/metrics_exporter_prometheus/struct.PrometheusBuilder.html#method.install
> When called from within a Tokio runtime, the exporter future is spawned directly into the runtime. Otherwise, a new single-threaded Tokio runtime is created on a background thread, and the exporter is spawned there.

At least a comment explaining the reasoning should be added.
It was requested during  the review from @divergentdave to use the existing runtime for the endpoint.

Reply:
Ahh, okay. Given that this is called after `runtime.enter()` is called, I think that's fine, but I agree that it would be useful to add comments about this. I actually think it would be helpful either way if we inline this code so that we have the related code nearby.